platforms/ide/problems-api/src/main/java/org/gradle/api/problems/deprecation/DeprecationReporter.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.problems.deprecation;
+
+import org.gradle.api.Action;
+import org.gradle.api.Incubating;
+import org.gradle.api.problems.deprecation.source.ReportSource;
+
+/**
+ * Specialized reporter for deprecation problems.
+ *
+ * @since 8.14
+ */
+@Incubating
+public interface DeprecationReporter {
+
+    /**
+     * Generic deprecation, using {@code message} to describe the deprecation.
```

Comment:
:x: I think the javadoc should be more elaborate on all methods. What's the expectation for consumers (reports, tapi clients, etc). Also, for `deprecateMethod`, the deprecation producer should know in what format the method should be passed. A concrete example would be very helpful.